.github/workflows/linux-publish.yml

```diff
@@ -0,0 +1,172 @@
+        id: parse
+        shell: bash
+        env:
+          GITHUB_REF_NAME: ${{ github.ref_name }}
+          GITHUB_RUN_ID: ${{ github.run_id }}
+        run: |
+          set -o pipefail
+          if [[ "${GITHUB_REF_NAME}" == main ]]
+          then
+            if PACKAGE_VERSION=$(jq -r .version package.json)
+            then
+              echo "version=${PACKAGE_VERSION#v}-${GITHUB_RUN_ID}" | tee -a $GITHUB_OUTPUT
+            else
+              echo "ERROR: Could not find version in package.json"
+              exit 1
+            fi
+          else
+            echo "version=${GITHUB_REF_NAME#app-ziti-console-v}" | tee -a $GITHUB_OUTPUT
+          fi
+
+  build-and-upload-app-ziti-console:
+    needs: parse-version
+    name: Build Console ZIP
+    runs-on: ubuntu-latest
+    env:
+      BUILD_CONFIG: "Release"
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          registry-url: https://npm.pkg.github.com/
+
+      - name: build app-ziti-console
+        shell: bash
+        run: ./build.sh /zac/
+
+      - name: upload app-ziti-console
+        uses: actions/upload-artifact@v4
+        with:
+          name: app-ziti-console
+          path: ./dist/app-ziti-console/
+
+  publish-linux-packages:
+    needs:
+      - parse-version
+      - build-and-upload-app-ziti-console
+    name: ${{ matrix.package_name }} ${{ matrix.arch.rpm }} ${{ matrix.nfpm_packager }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package_name:
+          - openziti-console
+        arch:
+          - goreleaser: amd64
+            deb: amd64
+            rpm: x86_64
+          - goreleaser: arm64
+            deb: arm64
+            rpm: aarch64
+          - goreleaser: armv7
+            deb: armv7
+            rpm: armv7
+        nfpm_packager:
+          - rpm
+          - deb
+          # - archlinux #  (pacman)
+    env:
+      ZAC_VERSION: ${{ needs.parse-version.outputs.version }}
+      ZITI_MAINTAINER: "OpenZiti Maintainers <developers@openziti.org>"
+      ZITI_HOMEPAGE: "https://openziti.io"
+      ZITI_VENDOR: "NetFoundry"
+      GOARCH: ${{ matrix.arch.goreleaser }}
+      ZITI_DEB_TEST_REPO: ${{ vars.ZITI_DEB_TEST_REPO || 'zitipax-openziti-deb-test' }}
+      ZITI_RPM_TEST_REPO: ${{ vars.ZITI_RPM_TEST_REPO || 'zitipax-openziti-rpm-test' }}
+      ZITI_DEB_PROD_REPO: ${{ vars.ZITI_DEB_PROD_REPO || 'zitipax-openziti-deb-stable' }}
+      ZITI_RPM_PROD_REPO: ${{ vars.ZITI_RPM_PROD_REPO || 'zitipax-openziti-rpm-stable' }}
+    steps:
+      - name: Checkout Workspace
+        uses: actions/checkout@v4
+
+      - name: Download app-ziti-console
+        uses: actions/download-artifact@v4
+        with:
+          name: app-ziti-console
+          path: ./dist/app-ziti-console
+
+      - name: Build Package
+        id: nfpm
+        uses: burningalchemist/action-gh-nfpm@v1
```

Comment:
this seems questionable. is there no other way to accomplish this?

Reply:
I've pinned the revision to eliminate the possibility that the `v1` tag could point to a malicious revision in the future.